include/seqan3/core/detail/strong_type.hpp

```diff
@@ -0,0 +1,546 @@
+ */
+#define SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(concept_name, operator_name)                                             \
+template <typename t>                                                                                                  \
+concept bool concept_name = requires(t v1, t v2)                                                                       \
+{                                                                                                                      \
+    { v1 operator_name v2 } -> t;                                                                                      \
+}
+
+/*!\brief Defines an internal concept to check the unary operator for the seqan3::detail::strong_type skills.
+ * \ingroup core
+ * \attention This concept definition is necessary, since gcc has a bug that prevents us from defining these
+ * requirements within the requires clause of the respective operator.
+ */
+#define SEQAN3_DEFINE_UNARY_OPERATOR_CONCEPT(concept_name, operator_name)                                              \
+template <typename t>                                                                                                  \
+concept bool concept_name = requires(t v1)                                                                             \
+{                                                                                                                      \
+    { operator_name v1 } -> t;                                                                                         \
+}
+
+//!\cond
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_add_concept,             +);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_subtract_concept,        -);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_multiply_concept,        *);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_divide_concept,          /);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_modulo_concept,          %);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_bitwise_and_concept,     &);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_bitwise_or_concept,      |);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_bitwise_xor_concept,     ^);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_bitwise_lshift_concept, <<);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_bitwise_rshift_concept, >>);
+SEQAN3_DEFINE_UNARY_OPERATOR_CONCEPT (_skill_bitwise_not_concept,     ~);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_logical_and_concept,    &&);
+SEQAN3_DEFINE_BINARY_OPERATOR_CONCEPT(_skill_logical_or_concept,     ||);
+SEQAN3_DEFINE_UNARY_OPERATOR_CONCEPT (_skill_logical_not_concept,     !);
+SEQAN3_DEFINE_UNARY_OPERATOR_CONCEPT (_skill_incrementable_concept,   ++);
+SEQAN3_DEFINE_UNARY_OPERATOR_CONCEPT (_skill_decrementable_concept,   --);
+//!\endcond
+
+ //!\cond
+template <typename strong_type, skill skills>
+struct skills_base;
+//!\endcond
+
+//!\cond
+template <typename unnamed_type, typename name_tag, skill skills>
```

Comment:
an "unnamed type" is something different:
https://stackoverflow.com/questions/5131691/what-is-an-unnamed-type-in-c
I would maybe use "underlying type" instead since we use that in other places, too.

Reply:
good point